git/client.go

```diff
@@ -376,59 +376,62 @@ func (c *Client) lookupCommit(ctx context.Context, sha, format string) ([]byte,
 	return out, nil
 }
 
-// ReadBranchConfig parses the `branch.BRANCH.(remote|merge|gh-merge-base)` part of git config.
+// ReadBranchConfig parses the `branch.BRANCH.(remote|merge|pushremote|gh-merge-base)` part of git config.
 // If no branch config is found or there is an error in the command, it returns an empty BranchConfig.
 // Downstream consumers of ReadBranchConfig should consider the behavior they desire if this errors,
 // as an empty config is not necessarily breaking.
 func (c *Client) ReadBranchConfig(ctx context.Context, branch string) (BranchConfig, error) {
 
 	prefix := regexp.QuoteMeta(fmt.Sprintf("branch.%s.", branch))
-	args := []string{"config", "--get-regexp", fmt.Sprintf("^%s(remote|merge|%s)$", prefix, MergeBaseConfig)}
+	args := []string{"config", "--get-regexp", fmt.Sprintf("^%s(remote|merge|pushremote|%s)$", prefix, MergeBaseConfig)}
 	cmd, err := c.Command(ctx, args...)
 	if err != nil {
 		return BranchConfig{}, err
 	}
 
-	out, err := cmd.Output()
+	// This is the error we expect if a git command does not run successfully.
+	// If the ExitCode is 1, then we just didn't find any config for the branch.
+	// We will use this error to check against the commands that are allowed to
+	// return an empty result.
+	var gitError *GitError
+	branchCfgOut, err := cmd.Output()
 	if err != nil {
-		// This is the error we expect if the git command does not run successfully.
-		// If the ExitCode is 1, then we just didn't find any config for the branch.
-		var gitError *GitError
 		if ok := errors.As(err, &gitError); ok && gitError.ExitCode != 1 {
 			return BranchConfig{}, err
 		}
 		return BranchConfig{}, nil
 	}
 
-	return parseBranchConfig(outputLines(out)), nil
+	return parseBranchConfig(outputLines(branchCfgOut)), nil
 }
 
-func parseBranchConfig(configLines []string) BranchConfig {
+func parseBranchConfig(branchConfigLines []string) BranchConfig {
 	var cfg BranchConfig
 
-	for _, line := range configLines {
+	// Read the config lines for the specific branch
+	for _, line := range branchConfigLines {
 		parts := strings.SplitN(line, " ", 2)
 		if len(parts) < 2 {
 			continue
 		}
 		keys := strings.Split(parts[0], ".")
 		switch keys[len(keys)-1] {
 		case "remote":
-			if strings.Contains(parts[1], ":") {
-				u, err := ParseURL(parts[1])
-				if err != nil {
-					continue
-				}
-				cfg.RemoteURL = u
-			} else if !isFilesystemPath(parts[1]) {
-				cfg.RemoteName = parts[1]
-			}
+			remoteURL, remoteName := parseRemoteURLOrName(parts[1])
+			cfg.RemoteURL = remoteURL
+			cfg.RemoteName = remoteName
+		// pushremote usually indicates a "triangular" workflow
+		case "pushremote":
+			pushRemoteURL, pushRemoteName := parseRemoteURLOrName(parts[1])
+			cfg.PushRemoteURL = pushRemoteURL
+			cfg.PushRemoteName = pushRemoteName
```

Comment:
Similar to the above, PR suggestions don't easily work across deletions:

```suggestion
		case "pushremote":
			cfg.PushRemoteURL, cfg.PushRemoteName := parseRemoteURLOrName(parts[1])
```